specs/ScreenCaptureRequested.md

```diff
@@ -0,0 +1,348 @@
+# Background
+
+JavaScript's Screen Capture API `GetDisplayMedia` allows developers to get a video stream of a 
+user's tabs, windows, or desktop. This API is available in WebView2, but the current default UI 
+has some problems that we need to fix to make sure that hybrid apps using WebView2 have a more 
+seamless web/native experience. This includes removing the tab column in the UI, replacing 
+default strings and icons that do not match in WV2, and potentially having the ability to 
+customize the UI itself. 
+
+These apps also expect that the screen capture dialog has an event before the UI is shown to give 
+the host app an opportunity to block or allow UI from showing at all.
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+# Description
+
+We propose introducing the `ScreenCaptureRequested` event. This event will be raised whenever 
+the WebView2 and/or iframe corresponding to the CoreWebView2Frame or any of its descendant iframes 
+requests permission to use the Screen Capture API before the UI is shown. 
+
+To maintain backwards compatibility, by default we plan to raise

```

Comment:
We aren't doing this for compat. This is a new event. We're adding it on both frame and webview2 for convenience of end dev.